### ISSUE TITLE
Custom breadcrumbs or context is not sent to sentry.

### ISSUE THREAD

**Issue by GeeWee** [getsentry/raven-node#396]
The code: 

```javascript
Raven.config(
      'https://****@sentry.io/233**',       
     {
         captureUnhandledRejections: true,
     },
);

Raven.context( () => {
    Raven.captureBreadcrumb({
        message: 'Test breadcrumb',
        category: 'payment',
        data: {
            amount: 312,
        }
    });

    Raven.setContext({
        context: 'someContext'
    });
    throw Error(' in context error 2 ')
});
```

Sends the error to sentry, but I'm unable to see the additional context anywhere, and the breadcrumbs aren't available either.

I'm using hosted sentry, and with a raven-node version of "^2.2.1" in my package.json. I'm setting up raven-node as shown above.

**Comment by kamilogorek** [comment]
Hey @GeeWee, there are few things going on here:

- you have to call `install` on `Raven` object in order to use breadcrumbs (after calling `.config`)
- `setContext` is as it's name says, setting a whole context – overriding it, and because breadcrumbs are also part of the error context, they are disappearing
- use `mergeContext` if you want to add some additional values without overriding it
- Sentry supports only specific context keys, `user`, `tags` and `extras` (which are any arbitrary values really) – please refer to https://docs.sentry.io/learn/context/

To summarize, here's a version of your code that will work:

```js
Raven.config(
      'https://****@sentry.io/233**',       
     {
         captureUnhandledRejections: true,
     },
).install();

Raven.context( () => {
    Raven.captureBreadcrumb({
        message: 'Test breadcrumb',
        category: 'payment',
        data: {
            amount: 312,
        }
    });

    Raven.mergeContext({
        extras: {
            moreValues: 'someContext'
        }
    });

    throw Error(' in context error 2 ')
});
```

**Comment by GeeWee** [comment]
You're absolutely right. Thanks a bunch. (Although for anyone else reading, it's `extra` - not `extras`)

I think the behaviour of the following code is a little strange:

```javascript
Raven.config(
  'https://****@sentry.io/**',
		{
			captureUnhandledRejections: true,
		},
	) 
//NOTE NO INSTALL call

Raven.context( () => {
	Raven.captureBreadcrumb({
		message: 'Test breadcrumb',
		category: 'payment',
		data: {
			amount: 312,
		}
	});
	
	Raven.mergeContext({
		extra: {
			moreValues: 'someContext'
		},
		user: {
			foo: 'bar',
		},
		tags: {tag1: 'tag2',}
	});
	
	throw Error(' raven error 3')
});
```
Note that there's only a config call, and no install call.
This flawlessly captures the context and the error, but no breadcrumbs. I think it's a little counterintuitive you can end up in a situation where half of raven works and the other half doesn't. A warning at least would be nice.
![image](https://user-images.githubusercontent.com/6381792/32828837-88baf320-c9f0-11e7-8054-587783e9063e.png)

Anyways - feel free to close this issue if you'd rather track this in another issue, or this is working as intended.

**Comment by israel-viveros** [comment]
i have sam error with node 8.6.0

**Comment by kamilogorek** [comment]
@GeeWee addressed in https://github.com/getsentry/raven-node/pull/402
@israel-viveros what error do you have exactly? Not capturing breadcrumbs without installing raven is intended behaviour for now. If there's anything else that causes issues for you, please open a new ticket. Thanks!
